### ISSUE TITLE
Register MPS-specific lowering for jax.nn.dot_product_attention

### ISSUE THREAD

**Issue by tillahoffmann** [tillahoffmann/jax-mps#125]
## Goal

Make `jax.nn.dot_product_attention(...)` automatically dispatch to our fused `mps.sdpa` / `mps.sdpa_causal` custom_call on the MPS platform — no user code changes required.

## Approach

Preferred: find the JAX primitive that `jax.nn.dot_product_attention` binds (recent JAX versions expose a \`dot_product_attention_p\` primitive with platform-specific lowerings for cuDNN on GPU) and register an MPS lowering:

\`\`\`python
mlir.register_lowering(dot_product_attention_p, _mps_dpa_lowering, platform=\"mps\")
\`\`\`

Same mechanism we already use for \`eigh_p\`, \`qr_p\`, \`svd_p\`, etc.

## Fallback

If no stable primitive exists in our supported JAX versions, consider monkeypatching \`jax.nn.dot_product_attention\`. Downsides:

- Affects all platforms in-process (needs runtime platform gating)
- Fragile across JAX versions — signature / kwargs have changed
- Surprising side effect of \`import jax_plugins.mps\` mutating \`jax.nn\`
- Won't catch hand-rolled \`softmax(Q @ K^T) @ V\` implementations

Prefer \`register_lowering\` if feasible.

## Out of scope

Intercepting hand-rolled attention — users who don't use \`jax.nn.dot_product_attention\` will keep the unfused path.

## References

- \`src/jax_plugins/mps/ops.py\` — existing \`mps.sdpa\` / \`mps.sdpa_causal\` primitives
- JAX source: \`jax/_src/nn/functions.py\` (location of primitive, version-dependent)

**Comment by tillahoffmann** [comment]
Related: #133 tracks an alternative approach — pattern-matching the hand-rolled `softmax(Q @ K^T / sqrt(d)) @ V` chain in the lowered StableHLO and rewriting it to `mps.sdpa`. That covers user code that doesn't go through `jax.nn.dot_product_attention` at all. The two approaches are complementary: `register_lowering` for the JAX-native entry point (this issue), pattern-matching for hand-rolled attention (#133).
